include/net/net_pkt.h

```diff
@@ -170,6 +174,23 @@ static inline void net_pkt_set_iface(struct net_pkt *pkt, struct net_if *iface)
 	pkt->lladdr_dst.type = iface->link_addr.type;
 }
 
+static inline struct net_if *net_pkt_orig_iface(struct net_pkt *pkt)
+{
+#if defined(CONFIG_NET_ROUTING)
+	return pkt->orig_iface;
+#else
+	return pkt->iface;
+#endif
+}
+
+static inline void net_pkt_set_orig_iface(struct net_pkt *pkt,
+					  struct net_if *iface)
+{
+#if defined(CONFIG_NET_ROUTING)
+	pkt->orig_iface = iface;
+#endif
```

Comment:
do as usual in this file: put the #ifdef around the whole fuction, and as an #else create a dummy macro that does nothing.

Reply:
Different people give different advice in reviews. Some have commented that they want to have inline functions that do nothing (like in this example), or then use macro that does nothing like you suggest here. I kind of like the dummy inline function better as it forces the compiler to check that compilation is ok. The end result is the same in both cases.

Reply:
Either way, my point was that it should be consistent within this file. So I guess you'll have to change all the places in that file that do not do this way.

Reply:
We can do that, but not in this PR.